components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/main/java/org/wso2/carbon/identity/api/resource/mgt/dao/impl/AuthorizationDetailsTypeMgtDAOImpl.java

```diff
@@ -0,0 +1,523 @@
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.wso2.carbon.identity.api.resource.mgt.util.AuthorizationDetailsTypesUtil.assertRichAuthorizationRequestsEnabled;
+import static org.wso2.carbon.identity.api.resource.mgt.util.AuthorizationDetailsTypesUtil.isRichAuthorizationRequestsDisabled;
+import static org.wso2.carbon.identity.api.resource.mgt.util.AuthorizationDetailsTypesUtil.parseSchema;
+import static org.wso2.carbon.identity.api.resource.mgt.util.AuthorizationDetailsTypesUtil.toJsonString;
+import static org.wso2.carbon.identity.api.resource.mgt.util.FilterQueriesUtil.getAuthorizationDetailsTypesFilterQueryBuilder;
+
+/**
+ * Implementation of the {@link AuthorizationDetailsTypeMgtDAO} interface for managing
+ * Authorization Detail Types in the database.
+ *
+ * <p>This class interacts with the persistence layer to perform CRUD operations on
+ * authorization detail types associated with APIs. It supports tenant-aware operations
+ * and ensures data isolation between tenants.</p>
+ */
+public class AuthorizationDetailsTypeMgtDAOImpl implements AuthorizationDetailsTypeMgtDAO {
+
+    private static final Log LOG = LogFactory.getLog(AuthorizationDetailsTypeMgtDAOImpl.class);
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<AuthorizationDetailsType> addAuthorizationDetailsTypes(
+            String apiId, List<AuthorizationDetailsType> authorizationDetailsTypes, Integer tenantId)
+            throws APIResourceMgtException {
+
+        try (Connection dbConnection = IdentityDatabaseUtil.getDBConnection(true)) {
+
+            final List<AuthorizationDetailsType> addedAuthorizationDetailsTypes =
+                    this.addAuthorizationDetailsTypes(dbConnection, apiId, authorizationDetailsTypes, tenantId);
+            IdentityDatabaseUtil.commitTransaction(dbConnection);
+
+            return addedAuthorizationDetailsTypes;
+        } catch (SQLException e) {
+            throw APIResourceManagementUtil.handleServerException(APIResourceManagementConstants.ErrorMessages
+                    .ERROR_CODE_ERROR_WHILE_ADDING_AUTHORIZATION_DETAILS_TYPES, e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<AuthorizationDetailsType> addAuthorizationDetailsTypes(
+            Connection dbConnection, String apiId, List<AuthorizationDetailsType> authorizationDetailsTypes,
+            Integer tenantId) throws APIResourceMgtException {
+
+        if (CollectionUtils.isEmpty(authorizationDetailsTypes)) {
+            return authorizationDetailsTypes;
+        }
+
+        assertRichAuthorizationRequestsEnabled();
```

Comment:
Move this to the service layer. checks like this should not be in the DAO layer.

Reply:
Fixed in 2d72c264576eb9e95bb40ef8fef70f8cdd5cbeb1